cardano_node_tests/tests/test_plutus_spend_build.py

```diff
@@ -653,6 +651,207 @@ def test_guessing_game(
+        ]
+        tx_output_fund = cluster.build_tx(
+            src_address=payment_addrs[0].address,
+            tx_name=f"{temp_template}_step1",
+            tx_files=tx_files_fund,
+            txouts=txouts_fund,
+            fee_buffer=2_000_000,
+            join_txouts=False,
+        )
+        tx_signed_fund = cluster.sign_tx(
+            tx_body_file=tx_output_fund.out_file,
+            signing_key_files=tx_files_fund.signing_key_files,
+            tx_name=f"{temp_template}_step1",
+        )
+
+        cluster.submit_tx(tx_file=tx_signed_fund, txins=tx_output_fund.txins)
+
+        txid_fund = cluster.get_txid(tx_body_file=tx_output_fund.out_file)
+        script_utxos1 = cluster.get_utxo(txin=f"{txid_fund}#1", coins=[clusterlib.DEFAULT_COIN])
+        script_utxos2 = cluster.get_utxo(txin=f"{txid_fund}#2", coins=[clusterlib.DEFAULT_COIN])
+        collateral_utxos1 = cluster.get_utxo(txin=f"{txid_fund}#3")
+        collateral_utxos2 = cluster.get_utxo(txin=f"{txid_fund}#4")
+
+        assert script_utxos1 and script_utxos2, "No script UTxOs"
+        assert collateral_utxos1 and collateral_utxos2, "No collateral UTxOs"
+
+        assert (
+            script_utxos1[0].amount == script_fund
+        ), f"Incorrect balance for script address `{script_utxos1[0].address}`"
+        assert (
+            script_utxos2[0].amount == script_fund
+        ), f"Incorrect balance for script address `{script_utxos2[0].address}`"
+
+        # Step 2: spend the "locked" UTxOs
+
+        assert plutus_op1.datum_file and plutus_op2.datum_file
+        assert plutus_op1.redeemer_cbor_file and plutus_op2.redeemer_cbor_file
+
+        plutus_txins = [
+            clusterlib.ScriptTxIn(
+                txins=script_utxos1,
+                script_file=plutus_op1.script_file,
+                collaterals=collateral_utxos1,
+                datum_file=plutus_op1.datum_file,
+                redeemer_cbor_file=plutus_op1.redeemer_cbor_file,
+            ),
+            clusterlib.ScriptTxIn(
+                txins=script_utxos2,
+                script_file=plutus_op2.script_file,
+                collaterals=collateral_utxos2,
+                datum_file=plutus_op2.datum_file,
+                redeemer_cbor_file=plutus_op2.redeemer_cbor_file,
+            ),
+        ]
+        tx_files_redeem = clusterlib.TxFiles(
+            signing_key_files=[payment_addrs[1].skey_file],
+        )
+        txouts_redeem = [
+            clusterlib.TxOut(address=payment_addrs[1].address, amount=amount * 2),
+        ]
+        tx_output_redeem = cluster.build_tx(
+            src_address=payment_addrs[0].address,
+            tx_name=f"{temp_template}_step2",
+            tx_files=tx_files_redeem,
+            txouts=txouts_redeem,
+            script_txins=plutus_txins,
+            change_address=payment_addrs[0].address,
+        )
+
+        # calculate cost of Plutus script
+        plutus_cost = cluster.calculate_plutus_script_cost(
```

Comment:
_suggestion:_ maybe we could move this to the part of the code that uses it line 833, like that we don't "interrupt" the logic: create the tx -> sign it -> submit it